### ISSUE TITLE
Gospel does not parse `fun` type annotations as OCaml

### ISSUE THREAD

**Issue by shym** [ocaml-gospel/gospel#305]
The following code is accepted by the OCaml type checker:

```ocaml
  fun true : int -> 12
```

(besides the obviously non-exhaustive pattern `true`), meaning that the `int` is the return type but the Gospel typechecker accepts the following:

```ocaml
val f : int list -> int
(*@ y = f xs
    ensures List.for_all (fun x : int -> true) xs *)
```

where `int` is interpreted as the type for the argument `x`.

**Comment by shym** [comment]
#309 will change that situation:
```
val f : int list -> int
(*@ y = f xs
    ensures List.for_all (fun x : int -> true) xs *)
```
will now report a syntax error at `: int` as the type annotation is not handled by the parser.

**Comment by n-osborne** [comment]
Fixed by #309
